cosmodc2/write_umachine_healpix_mock_to_disk.py

```diff
@@ -352,9 +363,9 @@ def write_umachine_healpix_mock_to_disk(
             if num_infinite > 0:
                 print('...Warning: {} infinite values in synthetic {}'.format(num_infinite, m_id))
 
-        #  Now downsample the synthetic galaxies according to the
+        #  Now downsample the synthetic galaxies to adjust for volume of lightcone shell
         #  desired number = synthetic_number*comoving_vol(snapshot)/comoving_vol(healpixel)
-        volume_factor = get_volume_factor(snapshot, redshift_max, z2ts)
+        volume_factor = get_volume_factor(redshift, previous_redshift, Vtotal, cosmology)
         num_selected_synthetic = int(synthetic_number*volume_factor)
```

Comment:
What happens here when volume_factor=0 ==> num_selected_synthetic=0?

Reply:
Yes, that happens in some edge cases. Sometimes the overlap of the edge pixel with the octant is tiny (eg the healpix boundary is at x~1e-14 and we want x>0). In this case, the overlap is too small to use the MonteCarlo method to estimate the overlap. I print a message that volume_factor=0 ==> num_selected_synthetic=0 and return an empty synthetics table. This is not perfect but it is a very peculiar edge case.